transfers_tel.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/moov-io/ach"
+	"github.com/moov-io/base"
+)
+
+type TELDetail struct {
+	PhoneNumber string `json:"phoneNumber"`
+}
+
+// createTELBatch creates and returns a TEL ACH batch for use after receiving oral authorization to debit a customer's account.
+//
+// TEL batches require a telephone number that's answered during typical business hours along with a date and statement of oral
+// authorization for a one-time funds transfer. Recurring transfers must contain the total amount of transfers or conditions for
+// scheduling transfers. Originators must retain written notice of the authorization for two years.
+func createTELBatch(id, userId string, transfer *Transfer, receiver *Receiver, receiverDep *Depository, orig *Originator, origDep *Depository) (ach.Batcher, error) {
+	batchHeader := ach.NewBatchHeader()
+	batchHeader.ID = id
+	batchHeader.ServiceClassCode = ach.DebitsOnly
+	batchHeader.CompanyName = orig.Metadata
+	batchHeader.StandardEntryClassCode = ach.TEL
+	batchHeader.CompanyIdentification = orig.Identification
+	batchHeader.CompanyEntryDescription = transfer.Description
+	batchHeader.EffectiveEntryDate = base.Now().AddBankingDay(1).Format("060102") // Date to be posted, YYMMDD
+	batchHeader.ODFIIdentification = aba8(origDep.RoutingNumber)
+
+	// Add EntryDetail to PPD batch
+	entryDetail := ach.NewEntryDetail()
+	entryDetail.ID = id
+	entryDetail.TransactionCode = determineTransactionCode(transfer, origDep)
+	entryDetail.RDFIIdentification = aba8(receiverDep.RoutingNumber)
+	entryDetail.CheckDigit = abaCheckDigit(receiverDep.RoutingNumber)
+	entryDetail.DFIAccountNumber = receiverDep.AccountNumber
+	entryDetail.Amount = transfer.Amount.Int()
+	entryDetail.IdentificationNumber = createIdentificationNumber() // TODO(adam): should this be the [required] phone number?
+	entryDetail.IndividualName = receiver.Metadata
+	entryDetail.DiscretionaryData = transfer.Description // TODO(adam): Or should this be the phone number
```

Comment:
I know for WEB it is S or R.  Does the same apply to TEL?

Reply:
Yea. PaymentTypeCode is S or R for TEL entries

http://docs.moov.io/en/latest/ach/file-structure/#tel-telephone-initiated-entry